tools/wptrunner/wptrunner/wpttest.py

```diff
@@ -149,12 +149,12 @@ def update_metadata(self, metadata=None):
     def from_manifest(cls, manifest_file, manifest_item, inherit_metadata, test_metadata):
         timeout = cls.long_timeout if manifest_item.timeout == "long" else cls.default_timeout
         protocol = "https" if hasattr(manifest_item, "https") and manifest_item.https else "http"
-        return cls(manifest_item.source_file.tests_root,
+        return cls(manifest_file.tests_root,
                    manifest_item.url,
                    inherit_metadata,
                    test_metadata,
                    timeout=timeout,
-                   path=manifest_item.source_file.path,
+                   path=os.path.join(manifest_file.tests_root, manifest_item.path),
```

Comment:
You could have a method on the manifest to return this

Reply:
That seems a bit weird to me, at least.